src/tracker.js

```diff
@@ -1,147 +1,158 @@
+      var track = tracks[i];
+      if (
+        track.active &&
+        (track.type === "video" || track.type === "variant")
+      ) {
+        return track;
       }
     }
-    return {}
+    return {};
   }
 
-  getLanguage () {
-    return this.getTrack().language
+  getLanguage() {
+    return this.getTrack().language;
   }
 
-  getRenditionName () {
-    return this.getTrack().label
+  getRenditionName() {
+    return this.getTrack().label;
   }
 
-  getRenditionBitrate () {
-    return this.getTrack().bandwidth
+  getRenditionBitrate() {
+    return this.getTrack().bandwidth;
   }
 
-  getRenditionWidth () {
-    return this.getTrack().width
+  getRenditionWidth() {
+    return this.getTrack().width;
   }
 
-  getRenditionHeight () {
-    return this.getTrack().height
+  getRenditionHeight() {
+    return this.getTrack().height;
   }
 
-  getPlayerVersion () {
-    return shaka.Player.version
+  getPlayerVersion() {
+    return shaka.Player.version;
   }
 
-  registerListeners () {
-    nrvideo.Log.debugCommonVideoEvents(this.tag)
+  registerListeners() {
+
+    nrvideo.Log.debugCommonVideoEvents(this.tag);
     nrvideo.Log.debugCommonVideoEvents(this.player, [
-      null, 'buffering', 'loading', 'adaptation', 'emsg', 'trackschanged', 'unloading',
-      'expirationupdated', 'largegap', 'texttrackvisibility', 'trackschanged'
-    ])
+      null,
+      "buffering",
+      "loading",
+      "adaptation",
+      "emsg",
+      "trackschanged",
+      "unloading",
+      "expirationupdated",
+      "largegap",
+      "texttrackvisibility",
+      "trackschanged",
+    ]);
 
-    this.tag.addEventListener('loadstart', this.onDownload.bind(this))
-    this.tag.addEventListener('loadedmetadata', this.onDownload.bind(this))
-    this.tag.addEventListener('loadeddata', this.onDownload.bind(this))
-    this.tag.addEventListener('play', this.onPlay.bind(this))
-    this.tag.addEventListener('playing', this.onPlaying.bind(this))
-    this.tag.addEventListener('pause', this.onPause.bind(this))
-    this.tag.addEventListener('seeking', this.onSeeking.bind(this))
-    this.tag.addEventListener('seeked', this.onSeeked.bind(this))
-    this.tag.addEventListener('error', this.onError.bind(this))
-    this.tag.addEventListener('ended', this.onEnded.bind(this))
+    this.tag.addEventListener("pause", this.onPause.bind(this));
+    this.tag.addEventListener("ended", this.onEnded.bind(this));
+    this.tag.addEventListener("play", this.onPlay.bind(this));
+    this.tag.addEventListener("loadedmetadata", this.onDownload.bind(this));
+    this.tag.addEventListener("loadeddata", this.onDownload.bind(this));
+    this.tag.addEventListener("loadstart", this.onDownload.bind(this));
+    this.tag.addEventListener("playing", this.onPlaying.bind(this));
+    this.tag.addEventListener("seeking", this.onSeeking.bind(this));
+    this.tag.addEventListener("seeked", this.onSeeked.bind(this));
+    this.tag.addEventListener("error", this.onError.bind(this));
 
```

Comment:
I haven't checked the documentation of Shaka, but is it possible to register all elements in the player directly or do we really need to use the media element (stored in the tag)? If we can rely only in the player, I think we should avoid storing and using the media element.

Reply:
This can be achieved but by doing this.player.getMediaElement().addEventlistner(). Because all events are not coming directly from player object.